Flaky evaluator tests

I have observed some flakyness in some of the evaluator tests. Will dump the stack traces here for now...

Found on ubuntu-latest, Python: 3.8

```
2021-06-10T06:22:40.3197383Z =================================== FAILURES ===================================
2021-06-10T06:22:40.3198263Z _____________________ test_run_and_cancel_legacy_ensemble ______________________
2021-06-10T06:22:40.3198816Z 
2021-06-10T06:22:40.3200581Z tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_run_and_cancel_legacy_ens0')
2021-06-10T06:22:40.3201623Z unused_tcp_port = 42933
2021-06-10T06:22:40.3202622Z make_ensemble_builder = <function make_ensemble_builder.<locals>._make_ensemble_builder at 0x7f694d5251f0>
2021-06-10T06:22:40.3203655Z 
2021-06-10T06:22:40.3204296Z     @pytest.mark.timeout(60)
2021-06-10T06:22:40.3205362Z     def test_run_and_cancel_legacy_ensemble(tmpdir, unused_tcp_port, make_ensemble_builder):
2021-06-10T06:22:40.3206213Z         num_reals = 10
2021-06-10T06:22:40.3206827Z         with tmpdir.as_cwd():
2021-06-10T06:22:40.3207636Z             ensemble = make_ensemble_builder(tmpdir, num_reals, 2).build()
2021-06-10T06:22:40.3209696Z             config = EvaluatorServerConfig(unused_tcp_port)
2021-06-10T06:22:40.3211552Z The script 'CsvExport2Job' caused an error while running:
2021-06-10T06:22:40.3212104Z     
2021-06-10T06:22:40.3213090Z             evaluator = EnsembleEvaluator(ensemble, config, 0, ee_id="1")
2021-06-10T06:22:40.3213928Z     
2021-06-10T06:22:40.3214774Z             with evaluator.run() as mon:
2021-06-10T06:22:40.3215596Z                 cancel = True
2021-06-10T06:22:40.3216241Z                 for _ in mon.track():
2021-06-10T06:22:40.3216857Z                     if cancel:
2021-06-10T06:22:40.3217517Z                         mon.signal_cancel()
2021-06-10T06:22:40.3218272Z                         cancel = False
2021-06-10T06:22:40.3218822Z     
2021-06-10T06:22:40.3219633Z >           assert evaluator._ensemble.get_status() == state.ENSEMBLE_STATE_CANCELLED
2021-06-10T06:22:40.3220881Z E           AssertionError: assert 'Stopped' == 'Cancelled'
2021-06-10T06:22:40.3222303Z E             - Cancelled
2021-06-10T06:22:40.3223032Z E             + Stopped
```

--- a/tests/ensemble_evaluator/conftest.py
+++ b/tests/ensemble_evaluator/conftest.py
@@ -77,7 +77,7 @@ def queue_config():
 
 @pytest.fixture
 def make_ensemble_builder(queue_config):
-    def _make_ensemble_builder(tmpdir, num_reals, num_jobs):
+    def _make_ensemble_builder(tmpdir, num_reals, num_jobs, job_sleep=0):
         builder = create_ensemble_builder()
         with tmpdir.as_cwd():
             ext_job_list = []
@@ -90,8 +90,11 @@ def make_ensemble_builder(queue_config):
                 with open(ext_job_exec, "w") as f:
                     f.write(
                         "#!/usr/bin/env python\n"
+                        "import time\n"
+                        "\n"
                         'if __name__ == "__main__":\n'
                         f'    print("stdout from {job_index}")\n'
+                        f"    time.sleep({job_sleep})\n"
                     )
 
                 ext_job_list.append(

--- a/tests/ensemble_evaluator/test_ensemble_legacy.py
+++ b/tests/ensemble_evaluator/test_ensemble_legacy.py
@@ -33,7 +33,7 @@ def test_run_legacy_ensemble(tmpdir, unused_tcp_port, make_ensemble_builder):
 def test_run_and_cancel_legacy_ensemble(tmpdir, unused_tcp_port, make_ensemble_builder):
     num_reals = 10
     with tmpdir.as_cwd():
-        ensemble = make_ensemble_builder(tmpdir, num_reals, 2).build()
+        ensemble = make_ensemble_builder(tmpdir, num_reals, 2, job_sleep=5).build()
         config = EvaluatorServerConfig(unused_tcp_port)
 
         evaluator = EnsembleEvaluator(ensemble, config, 0, ee_id="1")

